roles/gluster-secure-shared-storage/tasks/restrict_shared_storage.yml

```diff
@@ -0,0 +1,58 @@
+---
+# vim: set ts=2 sw=2 et :
+
+- name: Check for required variables
+  fail: msg="Variable {{ item }} must be defined to use this playbook"
+  when: item is undefined
+  with_items:
+  - cluster_master
+  - cluster_member_group
+
+- run_once: true
+  delegate_to: "{{ cluster_master }}"
+  block:
+  - name: Check if all hosts that form the gluster cluster are part of the play
+    fail: msg="All hosts that form the gluster cluster are not part of the play, missing {{ gl_gluster_hosts }}"
+    when: gl_gluster_hosts|length != 0
+    vars:
+      gl_gluster_hosts: "{{ groups[hostvars[inventory_hostname].cluster_member_group] | difference(play_hosts) }}"
+
+  - name: Check if additional hosts that are not part of the gluster cluster are not part of the play
+    fail: msg="More hosts than ones that form the gluster cluster are part of the play, additional {{ gl_this_play_hosts }}"
+    when: gl_this_play_hosts|length != 0
+    vars:
+      gl_this_play_hosts: "{{ play_hosts | difference(groups[hostvars[inventory_hostname].cluster_member_group]) }}"
+
+- name: Stop the gluster shared storage mounts
```

Comment:
Does this work on hosts that were not part of the original 3-nodes that host shared_storage? I'm not sure the systemd unit gets enabled on the addl hosts. I've always done this via a straight unmount call (see gluster-volume where it sets up shared_storage)

Reply:
The systemd unit comes with glusterfs-server package. So on systems that do not have this installed, it will not work. (file: /usr/lib/systemd/system/glusterfssharedstorage.service )

By additional hosts, you mean hosts that are part of the cluster but, are not part of any volume? Or, hosts that are not part of the cluster?

For the latter, this play wont even run, right?

For the former, glusterfs-server would be installed (as glusterd is part of the same package) and hence this would work as expected.

The systemd unit being enabled is an interesting observation. In my setup, before writing the ansible, when experimenting I did see the systemd unit stating the mount was inactive, and as I progressed through the various commands, and started using systemd to manage the service, things settled down. Not sure if this is a problem and why it occurs. (checked on the production systems and current status is as it was initially in my test system).

Reply:
Based on observations, systemd may not be managing this instance of the service, and hence trusting that is not safe. Moving ahead with suggestions as per the review comment.